src/lit-html.ts

```diff
@@ -343,15 +339,17 @@ export class NodePart extends Part {
     }
   }
 
-  clear() {
+  clear(startNode :Node = this.startNode) {
     this._previousValue = undefined;
-    const range = document.createRange();
-    range.setStartAfter(this.startNode);
-    range.setEndBefore(this.endNode);
-    range.deleteContents();
-    range.detach();
-  }
 
+    let node = startNode.nextSibling!;
+
+    while (node && node !== this.endNode) {
```

Comment:
I'm trying to do explicit checks for `undefined` and `null` in here, especially in loops, on advice from the V8 team which says that it's faster by avoiding a ToBoolean conversion. I'm also using `===`, and relying somewhat on TypeScript (which has very good typings for the DOM) to yell when I check for the wrong nully value. So:
```ts
while (node !== null && node !== this.endNode) {
```